openmc/deplete/microxs.py

```diff
@@ -192,8 +209,93 @@ def __init__(self, data: np.ndarray, nuclides: List[str], reactions: List[str]):
         self._index_nuc = {nuc: i for i, nuc in enumerate(nuclides)}
         self._index_rx = {rx: i for i, rx in enumerate(reactions)}
 
-    # TODO: Add a classmethod for generating MicroXS directly from cross section
-    # data using a known flux spectrum
+    @classmethod
+    def from_multi_group_flux(
+        cls,
+        energies: Union[Iterable[float], str],
+        multi_group_flux: Sequence[float],
```

Comment:
What's the rationale for using `Sequence` instead of `List`?

Reply:
We use Sequence in a few other places in openmc and one advantage is that Sequence allows tuples and lists. This allows us to pass in non mutable tuples.